File: dataset/human/java/file_9669.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Node[] nodes = new Node[n];
        for (int i = 0; i < n; i++) {
            int id = sc.nextInt();
            int[] children = new int[]{sc.nextInt(), sc.nextInt()};
            nodes[id] = new Node(id, children);
        }

        solve(nodes);

        for (int i = 0; i < nodes.length; i++) {
            Node node = nodes[i];
            int degree = 0;
            for (int j = 0; j < node.children.length; j++) {
                if (node.children[j] != -1) {
                    degree++;
                }
            }
            String type = node.type == Type.INTERNAL_NODE ? "internal node" : node.type == Type.LEAF ? "leaf" : "root";
            System.out.println("node " + node.id + ": parent = " + node.parent + ", sibling = " + node.sibling
                               + ", degree = " + degree + ", depth = " + node.depth + ", height = " + node.height
                               + ", " + type);
        }
    }

    private static void solve(Node[] nodes) {
        for (int i = 0; i < nodes.length; i++) {
            Node n = nodes[i];
            if (n.children[0] != -1) {
                nodes[n.children[0]].parent = n.id;
                nodes[n.children[0]].sibling = n.children[1];
            }
            if (n.children[1] != -1) {
                nodes[n.children[1]].parent = n.id;
                nodes[n.children[1]].sibling = n.children[0];
            }
        }

        int root = -1;
        for (int i = 0; i < nodes.length; i++) {
            Node n = nodes[i];
            if (n.parent == -1) {
                root = n.id;
                n.type = Type.ROOT;
                n.depth = 0;
            }
        }

        nodes[root].height = rec(nodes, root, 1);
    }

    private static int rec(Node[] nodes, int parent, int depth) {
        int[] children = nodes[parent].children;
        int height = -1;
        for (int i = 0; i < children.length; i++) {
            if (children[i] != -1) {
                Node node = nodes[children[i]];
                node.depth = depth;
                node.height = rec(nodes, children[i], depth + 1);
                height = Math.max(height, node.height);
                if (node.height != 0) {
                    node.type = Type.INTERNAL_NODE;
                } else {
                    node.type = Type.LEAF;
                }
            }
        }
        return height + 1;
    }

    static class Node {

        Node(int id, int[] children) {
            this.id = id;
            this.children = children;
            this.parent = -1;
            this.sibling = -1;
        }

        int id;
        int sibling;
        int parent;
        int depth;
        int height;
        int[] children;
        Type type;
    }

    enum Type {
        ROOT, INTERNAL_NODE, LEAF;
    }
}

File: dataset/human/c/file_9669.c
#include<stdio.h>

#define NIL -1

int depth(int);

struct Node{
  int parent;
  int left;
  int right;
};

struct Node T[100000];

int main(){

  int n,id,k,c[100000],i,j,m,de,count=0;

  scanf("%d",&n);

  for(i=0;i<n;i++){
    T[i].parent=NIL;
    T[i].right=NIL;
    T[i].left=NIL;
  }

  while(1){

    scanf("%d%d",&id,&k);
    count++;

    
    // printf("wa-i1\n");

    for(j=0;j<k;j++){
      
      scanf("%d",&c[j]);
    }

    
    /* if(count==1){
      
      T[id].parent=NIL;
      T[id].left=c[0];
      T[id].right=NIL;
      //printf("T:%d c[0]:%d\n",T[id].left,c[0]);
      } */
    
    if(k!=0){
      
      T[id].left=c[0];
      
      for(j=0;j<k;j++){
	T[c[j]].parent=id;
	
	if(j==k-1){T[c[j]].right=NIL;} 
	else {T[c[j]].right=c[j+1];}
	
      }
    }
    
    /* else if(k==0){
      
      T[id].left=NIL;
      
      
      // T[id].right=NIL;
	 }*/

    if(count==n)break;
  }

  //fprintf(stderr,"wa-i2\n");
  
  for(i=0;i<n;i++){

    //printf("wa-i4\n");
    
    de = depth(i);

    //printf("wa-i3\n");
    
    printf("node %d: parent = %d, depth = %d,",i,T[i].parent,de);
    
    if(T[i].parent==NIL){printf(" root, ");}
    else if(T[i].left==NIL){printf(" leaf, ");}
    else {printf(" internal node, ");}
    
    printf("[");

    j=T[i].left;

    //printf("\nj:%d\n",j);
    
    while(j!=NIL){

      if(T[i].left==j){
	printf("%d",j);
       
      }
      else{
	printf(", %d",j);
      }
      j=T[j].right;
    }
    
    printf("]\n");
  }

  //printf("node1:right:%d\n",T[].right);


  
  return 0;
}

int depth(int num){

  if(T[num].parent==NIL)return 0;

  return depth(T[num].parent)+1;
  
}
